src/main/java/com/minecolonies/core/client/gui/townhall/WindowMainPage.java

```diff
@@ -203,10 +171,7 @@ public String getLabel(final int index)
      */
     private void toggleTexture(final DropDownList dropDownList)
```

Comment:
I remember we had those ifs so that it doesn't toggle on UI open already or so. (no need to toggle for same as selected)

Reply:
I fixed that by actually calling the methods in the correct order during setup.
-> First assign data provider
-> Set selected index
-> Set update handler

Reply:
Also, the if statements had a negative side effect, because changing back to the initially selected item was not possible because the final values did not update, so that's another bug fixed